packages/slice-machine/lib/models/common/ComponentUI.ts

```diff
@@ -78,9 +78,33 @@ export const ComponentUI = {
       __status: computeStatus(component, remoteSlices),
     };
   },
+  variation(
+    component: ComponentUI,
+    variationId?: string
+  ): VariationSM | undefined {
+    if (component.model.variations.length) {
+      if (variationId)
+        return component.model.variations.find((v) => v.id === variationId);
+      return component.model.variations[0];
+    }
+  },
+
+  updateVariation(component: ComponentUI, variationId: string) {
+    return (mutate: (v: VariationSM) => VariationSM): ComponentUI => {
+      const variations = component.model.variations.map((v) => {
+        if (v.id === variationId) return mutate(v);
```

Comment:
what does `mutate` do?

Reply:
mutate is the a function parameter of the function (passed in line 93). In this case it would be the `v => {...}` 
<img width="463" alt="image" src="https://user-images.githubusercontent.com/69144671/179490239-f05a8497-2800-4879-9ae0-25c6b8000c09.png">

It's probably badly named, will change it to mutateCallbackFn

Reply:
does it do any mutations?